packages/middleware-flexible-checksums/src/flexibleChecksumsMiddleware.spec.ts

```diff
@@ -0,0 +1,145 @@
+import { getChecksum } from "./getChecksum";
+import { getChecksumAlgorithmForRequest } from "./getChecksumAlgorithmForRequest";
+import { getChecksumLocationName } from "./getChecksumLocationName";
+import { FlexibleChecksumsMiddlewareConfig } from "./getFlexibleChecksumsPlugin";
+import { hasHeader } from "./hasHeader";
+import { FlexibleChecksumsResolvedConfig } from "./resolveFlexibleChecksumsConfig";
+import { selectChecksumAlgorithmFunction } from "./selectChecksumAlgorithmFunction";
+import { validateChecksumFromResponse } from "./validateChecksumFromResponse";
+
+jest.mock("@aws-sdk/protocol-http");
+jest.mock("./getChecksumAlgorithmForRequest");
+jest.mock("./getChecksumLocationName");
+jest.mock("./selectChecksumAlgorithmFunction");
+jest.mock("./getChecksum");
+jest.mock("./hasHeader");
+jest.mock("./validateChecksumFromResponse");
+
+describe(flexibleChecksumsMiddleware.name, () => {
+  const mockNext = jest.fn();
+
+  const mockChecksum = "mockChecksum";
+  const mockChecksumAlgorithmFunction = jest.fn();
+  const mockChecksumLocationName = "mock-checksum-location-name";
+
+  const mockInput = {};
+  const mockConfig = {} as FlexibleChecksumsResolvedConfig;
+  const mockMiddlewareConfig = { input: mockInput } as FlexibleChecksumsMiddlewareConfig;
+
+  const mockBody = {};
+  const mockHeaders = {};
+  const mockRequest = { body: mockBody, headers: mockHeaders };
+  const mockArgs = { request: mockRequest } as BuildHandlerArguments<any>;
+  const mockResult = { response: {} };
+
+  beforeEach(() => {
+    mockNext.mockResolvedValueOnce(mockResult);
+    const { isInstance } = HttpRequest;
+    (isInstance as unknown as jest.Mock).mockReturnValue(true);
+    (getChecksumAlgorithmForRequest as jest.Mock).mockReturnValue(ChecksumAlgorithm.MD5);
+    (getChecksumLocationName as jest.Mock).mockReturnValue(mockChecksumLocationName);
+    (selectChecksumAlgorithmFunction as jest.Mock).mockReturnValue(mockChecksumAlgorithmFunction);
+    (getChecksum as jest.Mock).mockReturnValue(mockChecksum);
+    (hasHeader as jest.Mock).mockReturnValue(false);
+  });
+
+  afterEach(() => {
+    expect(mockNext).toHaveBeenCalledTimes(1);
+    jest.clearAllMocks();
+  });
+
+  describe("skips", () => {
+    it("if not an instance of HttpRequest", async () => {
+      const { isInstance } = HttpRequest;
+      (isInstance as unknown as jest.Mock).mockReturnValue(false);
+      const handler = flexibleChecksumsMiddleware(mockConfig, mockMiddlewareConfig)(mockNext, {});
+      await handler(mockArgs);
+      expect(getChecksumAlgorithmForRequest).not.toHaveBeenCalled();
+    });
+
+    describe("request checksum", () => {
+      afterEach(() => {
+        expect(getChecksumAlgorithmForRequest).toHaveBeenCalledTimes(1);
+        expect(selectChecksumAlgorithmFunction).not.toHaveBeenCalled();
+        expect(getChecksum).not.toHaveBeenCalled();
+      });
+
+      it("if checksumAlgorithm is not defined", async () => {
```

Comment:
nit: document the expect behavior for the cases? Same as other test cases.
```suggestion
      it("should not compute checksum if checksumAlgorithm is not defined", async () => {
```

Reply:
The describe block already calls this out.

I updated to make it explicit in [`199944c` (#3340)](https://github.com/aws/aws-sdk-js-v3/pull/3340/commits/199944c71d354a84f0d03bd44b2b8c8296961fa0)